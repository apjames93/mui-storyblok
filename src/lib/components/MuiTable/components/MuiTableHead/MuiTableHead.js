import React from 'react';
import PropTypes from 'prop-types';
import { TableHead } from '@material-ui/core';
import MuiTableRow from '../MuiTableRow/MuiTableRow';
import StoryBlok from '../../../../utils/Storyblok';
import { validComponents } from '../../../../utils/customProps';
import { renderComponents } from '../../../../utils/customComponents';

const MuiTableHead = ({
  rootClass,
  content,
}) => {
  const styles = StoryBlok.arrayToMuiStyles(rootClass);

  return (
    <TableHead className={styles.root}>
      {content.map((component, key) => renderComponents({ MuiTableRow }, component, key))}
    </TableHead>
  );
};

export default MuiTableHead;

MuiTableHead.propTypes = {
  /**
   * stroyblok multiselect of css classes
   * Mui Override or extend the styles applied to the component.
   */
  rootClass: PropTypes.arrayOf(PropTypes.string),

  /**
   * Content passed to render
   * components:
    MuiTableRow,
   */
  content(props, propName, componentName) {
    const components = ['MuiTableRow'];
    return validComponents(props, propName, componentName, components);
  },
};

MuiTableHead.defaultProps = {
  rootClass: [],
  content: [],
};
