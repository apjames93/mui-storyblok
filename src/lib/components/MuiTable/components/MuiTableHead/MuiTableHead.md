##### MuiTableHead has the ability to render custom components.

MuiTableHead small example:

```js
const tHead = [{
  component: 'MuiTableRow',
  content: [{
    component: 'MuiTableCell',
    size: 'small',
    content: [{
      component: 'MuiTypography',
      content: [{
        component: 'MuiText',
        content: 'Table Header 1',
      }],
    }],
  },
  {
    component: 'MuiTableCell',
    size: 'small',
    content: [{
      component: 'MuiTypography',
      content: [{
        component: 'MuiText',
        content: 'Table Header 2',
      }],
    }],
  },
  {
    component: 'MuiTableCell',
    size: 'small',
    content: [{
      component: 'MuiTypography',
      content: [{
        component: 'MuiText',
        content: 'Table Header 3',
      }],
    }],
  }],
}];

<MuiTableHead
  content={tHead}
/>
```

MuiTableHead large example:

```js
const tHead = [{
  component: 'MuiTableRow',
  content: [{
    component: 'MuiTableCell',
    size: 'large',
    content: [{
      component: 'MuiTypography',
      content: [{
        component: 'MuiText',
        content: 'Table Header 1',
      }],
    }],
  },
  {
    component: 'MuiTableCell',
    size: 'large',
    content: [{
      component: 'MuiTypography',
      content: [{
        component: 'MuiText',
        content: 'Table Header 2',
      }],
    }],
  },
  {
    component: 'MuiTableCell',
    size: 'large',
    content: [{
      component: 'MuiTypography',
      content: [{
        component: 'MuiText',
        content: 'Table Header 3',
      }],
    }],
  }],
}];

<MuiTableHead
  content={tHead}
/>
```