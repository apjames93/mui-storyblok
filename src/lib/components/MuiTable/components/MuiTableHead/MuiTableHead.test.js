import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import MuiTableHead from './MuiTableHead';

function setup() {
  const props = {
    _editable: '<!--#storyblok#{"name": "MuiTableHead", "space": "48408", "uid": "947be9f0-47c3-4315-a95a-550f0c560eb5", "id": "307934"}-->',
    content: [
      {
        _editable: '<!--#storyblok#{"name": "MuiTableRow", "space": "48408", "uid": "947be9f0-47c3-4315-a95a-550f0c560eb5", "id": "307934"}-->',
        component: 'MuiTableRow',
        content: [
          {
            _editable: '<!--#storyblok#{"name": "MuiTableCell", "space": "48408", "uid": "947be9f0-47c3-4315-a95a-550f0c560eb5", "id": "307934"}-->',
            component: 'MuiTableCell',
            content: [
              {
                _editable: '<!--#storyblok#{"name": "MuiTypography", "space": "48408", "uid": "947be9f0-47c3-4315-a95a-550f0c560eb5", "id": "307934"}-->',
                component: 'MuiTypography',
                content: [
                  {
                    _editable: '<!--#storyblok#{"name": "MuiText", "space": "48408", "uid": "947be9f0-47c3-4315-a95a-550f0c560eb5", "id": "307934"}-->',
                    component: 'MuiText',
                    text: 'wooo',
                  },
                ],
              },
            ],
          },
        ],
      },
    ],
  };

  const comp = shallow(<MuiTableHead {...props} />);
  return { comp, props };
}

describe('<MuiTableHead />', () => {
  test('snapshot', () => {
    const { props } = setup();
    const tree = renderer.create(<MuiTableHead {...props} />);
    expect(tree).toMatchSnapshot();
  });
  it('should render MuiTableHead', () => {
    const { comp } = setup();
    expect(comp).toBeDefined();
  });
});
