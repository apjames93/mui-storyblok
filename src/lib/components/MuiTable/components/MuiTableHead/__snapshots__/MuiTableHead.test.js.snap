// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MuiTableHead /> snapshot 1`] = `
<thead
  className="MuiTableHead-root makeStyles-root-2"
  role={null}
>
  <tr
    className="MuiTableRow-root makeStyles-root-3 MuiTableRow-head"
    role={null}
  >
    <th
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-head makeStyles-root-4"
      scope="col"
    >
      <div
        className="MuiTypography-root makeStyles-root-5 MuiTypography-body1"
      >
        <span
          className="MuiTypography-root makeStyles-root-6 MuiTypography-body1"
        >
           undefined 
        </span>
      </div>
    </th>
  </tr>
</thead>
`;
